tools/check-links

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2020  Matthew Glazar
+# See end of file for extended copyright information.
+
+import argparse
+import re
+import sys
+from concurrent.futures import ThreadPoolExecutor as Pool
+from urllib.parse import urldefrag, urljoin, urlparse
+
+import requests
+from bs4 import BeautifulSoup
+
+
+allow_mails = [
+    "strager.nds@gmail.com",
+    "nicolas@petton.fr",
+    "lewing@isc.tamu.edu",
+    "don.h@free.fr",
+    "Bram@vim.org",
+    "maintainer@vim.org",
+    "marketing@jetbrains.com",
+]
+
+allowed_file_ext_for_soup = [
+    "html",
+]
+
+headers = {
+    "User-Agent": "Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2228.0 Safari/537.3"
+}
+
+
+def check_fragment(soup, fragment) -> bool:
+    if fragment:
+        return True if soup.select(f"#{fragment}") else False
+    return True
+
+
+def is_mailto_link(link) -> bool:
+    return True if re.search(r"^mailto:*", link) else False
+
+
+def check_mail_link(link: str, parent: str) -> None:
+    mail = link.partition(":")[-1]
+    if mail not in allow_mails:
+        print(f"(unknown mail) {parent}, {mail}")
```

Comment:
Almost-blocking: If an invalid email is found, this script will still exit successfully (code 0). I think we should call `Crawler.report_error` here instead of `print`ing directly.